reference-architecture/osp-cli/generate_inventory.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/sh
+OCP3_DOMAIN=${OCP3_DOMAIN:-ocp3.example.com}
+OCP3_CONTROL_DOMAIN=${CONTROL_DOMAIN:-control.${OCP3_DOMAIN}}
+MASTER_COUNT=${MASTER_COUNT:-3}
+INFRA_NODE_COUNT=${INFRA_NODE_COUNT:-2}
+APP_NODE_COUNT=${APP_NODE_COUNT:-3}
+
+INVENTORY=${INVENTORY:-inventory}
+
+M=$(($MASTER_COUNT-1))
+I=$(($INFRA_NODE_COUNT-1))
+A=$(($APP_NODE_COUNT-1))
+
+cat <<EOF > $INVENTORY
+[OSEv3:children]
+masters
+etcd
+nodes
+
+[masters]
+master-[0:${M}].${OCP3_CONTROL_DOMAIN} openshift_public_hostname=master-[0:${M}].${OCP3_DOMAIN}
+
+[masters:vars]
+openshift_schedulable=false
+openshift_router_selector="region=infra"
+openshift_registry_selector="region=infra"
+
+[etcd]
+master-[0:${M}].${OCP3_CONTROL_DOMAIN} openshift_public_hostname=master-[0:${M}].${OCP3_DOMAIN}
+
+[infra-nodes]
```

Comment:
We don't have different groups for different node types but I like it, so IDK if we should modify the documentation instead.